validations/src/validations.rs

```diff
@@ -1288,10 +1297,27 @@ pub fn update_utxo_diff(
     outputs: Vec<&ValueTransferOutput>,
     tx_hash: Hash,
 ) {
+    let mut input_pkh = if let Some(first) = inputs.first() {
+        utxo_diff.get(first.output_pointer()).map(|vt| vt.pkh)
+    } else {
+        None
+    };
```

Comment:
You know how much I love one-liners:
```suggestion
    let mut input_pkh = inputs.first().and_then(|first| utxo_diff.get(first.output_pointer()).map(|vt| vt.pkh));
```